File: leetcode/group2/657.py
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        point = [0, 0]
        for move in moves:
            if move == 'U':
                point[1] += 1
            elif move == 'D':
                point[1] -= 1
            elif move == 'L':
                point[0] -= 1
            elif move == 'R':
                point[0] += 1
        if point[0] == 0 and point[1] == 0:
            return True
        return False

File: leetcode/花花酱题集/657.js
/**
 * @param {string} moves
 * @return {boolean}
 */
var judgeCircle = function (moves) {
  const ori = [0, 0];
  const steps = {
    U: [0, 1],
    R: [1, 0],
    D: [0, -1],
    L: [-1, 0],
  };
  for (const direction of moves) {
    const step = steps[direction];
    ori[0] += step[0];
    ori[1] += step[1];
  }

  return ori[0] === 0 && ori[1] === 0;
};

const moves = "LL";
console.log(judgeCircle(moves));
